structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+""" Specific class section implementation"""
+
+from __future__ import annotations
+
+import typing as t
+import structuralcodes.core._section_results as s_res
+from shapely import Polygon
+from structuralcodes.sections._generic import GenericSection
+from structuralcodes.geometry import SurfaceGeometry
+from structuralcodes.materials.concrete import Concrete
+
+class RectangularSection(GenericSection): 
+    """This is the specific implementation of a rectangular class section.
+
+    The section is a 2D geometry where Y axis is horizontal while Z axis is
+    vertical.
+
+    The moments and curvatures around Y and Z axes are assumed positive
+    according to RHR.
+
+    Attributes: TODO
+        geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry of
+            the section.
+        name (str): The name of the section.
+        section_calculator (GenericSectionCalculator): The object responsible
+            for performing different calculations on the section (e.g. bending
+            strength, moment curvature, etc.).
+    """
+
+    def __init__(self, height: float, width: float, concrete: Concrete, name: t.Optional[str] = None, integrator: t.Literal['marin', 'fiber'] = 'marin') -> None:
+        """Initialize a GenericSection.
+
+        Arguments:
+            geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry
+                of the section.
+            name (str): The name of the section.
+            integrator (str): The name of the SectionIntegrator to use.
+        """
+        poly: Polygon = Polygon(((0, 0), (width, 0), (width, height), (0, height)))
+        geometry: SurfaceGeometry = SurfaceGeometry(poly, concrete)
+        geometry.translate(-width/2, -height/2)
+        if name is None:
+            name: str = 'RectangularSection'
+        super().__init__(geometry, name, integrator)
+
+        self.height: float = height
+        self.width: float = width
+
+        # Calculating gross section properties
+        # Gross area
+        area: float = height * width
+        perimeter: float = 2 * (height + width)
+        ea: float = area * geometry.material.get_tangent(eps=0)[0]
+        for geo in self.geometry.geometries:
+            pass
+        for geo in self.geometry.point_geometries:
+            pass
+
+        if geometry.density is not None:
+            mass = area * geometry.density * 1e-9
+        else:
+            mass = 0
+
+
+        gp = s_res.GrossProperties()
+        gp.area = area
+        gp.perimeter = perimeter
+        gp.ea = ea
+        gp.mass = mass
+        # TODO: more gross sectional properties, and store them in self._gross_properties
+
+    @property
+    def area(self) -> float:
+        """Returns Area of the cross section.
+        Only the gross area of concrete is computed.
+
+        Returns:
+            float: The area of concrete section.
+        """
+        return self.width * self.height
```

Comment:
This can be removed for now. Here we have anyhow an example of something that in the future can be computed in a `RectangularSectionCalculator` for efficiency reasons. Instead of integrating with fibers or marin, we just multiply `height` and `width`.

Reply:
You did not change or remove the code here.
It could be fine for me to leave it like that as soon as it is coherent with the base class. In particular it seems to me that we don't have an area property in the current version of `GenericSection` but the area itself is inside the gross_properties that re computed with the section calculator....

I would remove this for now...